stan/math/prim/prob/beta_binomial_cdf.hpp

```diff
@@ -87,46 +87,37 @@ return_type_t<T_size1, T_size2> beta_binomial_cdf(const T_n& n, const T_N& N,
     const T_partials_return N_dbl = value_of(N_vec[i]);
     const T_partials_return alpha_dbl = value_of(alpha_vec[i]);
     const T_partials_return beta_dbl = value_of(beta_vec[i]);
-
+    const T_partials_return N_minus_n = N_dbl - n_dbl;
     const T_partials_return mu = alpha_dbl + n_dbl + 1;
-    const T_partials_return nu = beta_dbl + N_dbl - n_dbl - 1;
+    const T_partials_return nu = beta_dbl + N_minus_n - 1;
+    const T_partials_return one = 1;
 
     const T_partials_return F
-        = F32((T_partials_return)1, mu, -N_dbl + n_dbl + 1, n_dbl + 2, 1 - nu,
-              (T_partials_return)1);
-
-    T_partials_return C = lgamma(nu) - lgamma(N_dbl - n_dbl);
-    C += lgamma(mu) - lgamma(n_dbl + 2);
-    C += lgamma(N_dbl + 2) - lgamma(N_dbl + alpha_dbl + beta_dbl);
-    C = exp(C);
+        = F32(one, mu, 1 - N_minus_n, n_dbl + 2, 1 - nu, one);
 
-    C *= F / stan::math::beta(alpha_dbl, beta_dbl);
-    C /= N_dbl + 1;
+    T_partials_return C = lbeta(nu, mu) - lbeta(alpha_dbl, beta_dbl)
+                          - lbeta(N_minus_n, n_dbl + 2);
+    C = F * exp(C) / (N_dbl + 1);
 
     const T_partials_return Pi = 1 - C;
 
     P *= Pi;
 
     T_partials_return dF[6];
-    T_partials_return digammaOne = 0;
-    T_partials_return digammaTwo = 0;
+    T_partials_return digammaDiff = 0;
```

Comment:
[optional]

This pattern of setting and then resetting is something we want to avoid for both clarity and efficiency.  As is, it gets set to 0 then reset if size1 or size2 is not constant.  The setting allocates memory, which resetting throws away without recovering.

A better pattern for this is:

```
T_partials_return digammaDiff
  = is_constant_all<T_size1, T_size2>::value ? 0
                                             : digamma(mu + nu) - digamma(alpha_dbl + beta_dbl);
```

It's not incorrect as written and the inefficiency is minor, so I'm marking this optional.

Reply:
[optional]

Rather than defining `digammaDiff = A - B` and then using `- digammaDiff`, it'd be more efficient to define `digammaDiff = B - A` and then use without the negation.

Reply:
Thanks, for the time being I'm merging as is, I've added them to my other branch in which I'm cleaning up code like this, so they won't get lost.